server/src/modules/user/user.service.ts

```diff
@@ -33,6 +33,12 @@ export class UserService {
     return mentees;
   }
 
+  async createUser(user: any): Promise<User> {
+    const pass = await PasswordHash.fromPassword(user.password);
+    user.passwordHash = pass;
+    return this.userModel.create(user);
```

Comment:
Note: you didn't remove the `password` field from `user`. I would call `delete user.password` after line `user.passwordHash = pass` to remove that key from `user`. It probably doesn't make a difference for now (since there's no `password` field on the User model), but it could potentially cause security issues in the future.